Automattic/woocommerce-payments
# Add E2E test to test checkout performance

#### Changes proposed in this Pull Request
We would like to create an E2E test to measure metrics such as first paint, first contentful paint, etc. The test will run 3 times to take an average. It measures 3 things on the `/checkout` page: 
- Stripe element checkout
- UPE checkout
- WooPay checkout

The PR uses a similar technique as woocommerce-blocks https://github.com/woocommerce/woocommerce-blocks/blob/trunk/tests/utils/performance.ts. 

#### Testing instructions
1. Follow E2E instructions here(https://github.com/Automattic/woocommerce-payments/tree/develop/tests/e2e) to setup.
2. Make sure your `tests/e2e/deps/wcp-dev-tools` folder is up-to-date. You can do `git pull` to make sure it is the latest.
3. Run `npm run test:e2e-performance`
4. You should see the test passed and a new file is created under `tests/e2e/reports/checkout-performance.txt`. Each line of the file is a JSON record of a performance metric. It should look something like this:
```
{"description":"Stripe element: Average","serverResponse":3693.6966666641333,"firstPaint":76.56499991814296,"domContentLoaded":89.84833319361012,"loaded":372.0683332843085,"firstContentfulPaint":76.56499991814296,"firstBlock":2955.526666715741}
{"description":"Stripe UPE: Average","serverResponse":4223.398333260168,"firstPaint":90.24666668847203,"domContentLoaded":123.69833327829838,"loaded":465.96499998122454,"firstContentfulPaint":90.24666668847203,"firstBlock":3817.1183334973953}
{"description":"WooPay: Average","serverResponse":4126.750000131627,"firstPaint":77.12333323433995,"domContentLoaded":95.77999993537863,"loaded":338.5383333079517,"firstContentfulPaint":77.12333323433995,"firstBlock":3170.6916666589677}
```

-------------------


**Post merge**
